core/src/main/java/org/apache/iceberg/catalog/BaseSessionCatalog.java

```diff
@@ -30,8 +30,10 @@
 import org.apache.iceberg.exceptions.NamespaceNotEmptyException;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.view.View;
+import org.apache.iceberg.view.ViewBuilder;
 
-public abstract class BaseSessionCatalog implements SessionCatalog {
+public abstract class BaseSessionCatalog implements SessionCatalog, ViewSessionCatalog {
```

Comment:
Doesn't adding `ViewSessionCatalog` here break compatibility with existing implementations? It will force them to implement the view methods. I'm surprised this wasn't picked up by revapi.

Reply:
in this case, maybe we should have a `BaseViewSessionCatalog` that extends `BaseSessionCatalog`? This would be similar to `BaseMetastoreCatalog` & `BaseMetastoreViewCatalog`.
Also I'm not sure why RevAPI didn't complain

Reply:
Yes, I think we should. And let's track down why RevAPI didn't complain! That seems dangerous.

Reply:
@nastra any update on why RevAPI didn't catch this?